evo_researcher/functions/research.py

```diff
@@ -36,15 +40,24 @@ def research(
     collection = create_embeddings_from_results(scraped, text_splitter, api_key=openai_key)
 
     vector_result_texts: list[str] = []
+    url_to_content_deemed_most_useful: dict[str, str] = {}
 
     for query in queries:
         top_k_per_query_results = collection.similarity_search(query, k=top_k_per_query)
         vector_result_texts += [result.page_content for result in top_k_per_query_results if result.page_content not in vector_result_texts]
+
+        for x in top_k_per_query_results:
+            url_to_content_deemed_most_useful[x.metadata["url"]] = x.page_content
```

Comment:
Looks like you could have >1 results in `top_k_per_query_results` with the same url. So I think you need to check for that and `+=` instead of overwriting.

Reply:
Oh 💩 Good catch! This is supposed to be 

```
url_to_content_deemed_most_useful[x.metadata["url"]] = x.metadata["content"]
```

`x.metadata["content"]` holds the whole scraped content, not just a chunk as `x.page_content`. Maybe with this fix my summary experiments will be better! 🙏